monitor/types/types.go

```diff
@@ -0,0 +1,75 @@
+package types
+
+import (
+	"context"
+	"io"
+
+	"github.com/containerd/console"
+	controllerapi "github.com/docker/buildx/controller/pb"
+	"github.com/moby/buildkit/client"
+)
+
+// Monitor provides APIs for attaching and controlling the buildx server.
+type Monitor interface {
```

Comment:
Ah, I guessed we might need something like this interface, nice :tada:

Since there's a lot of shared methods here with `BuildxController` could we have this interface extend that one? Then we just need to define the methods on top of that.

Reply:
`Monitor` hides `ref` arguments from `BuildxController` and the function signatures are different so it seems that we can't use `BuildxController`'s API for `Monitor`.

Reply:
Thank you, SGTM.

Reply:
Cherry-picked https://github.com/docker/buildx/commit/8063632c547f206f9fdc66809c5eb41886f666e3 to this PR.